atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -0,0 +1,254 @@
+    docker daemon has no knowledge of the built image.
+    """
+
+    EXCLUDE_FILENAME = 'exclude-platform'
+
+    key = 'orchestrate_build'
+
+    def __init__(self, tasker, workflow, platforms, build_kwargs,
+                 osbs_client_config):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param platforms: list<str>, platforms to build
+        :param build_kwargs: dict, keyword arguments for starting worker builds
+        :param osbs_client_config: str, path to directory containing osbs.conf
+        """
+        super(OrchestrateBuildPlugin, self).__init__(tasker, workflow)
+        self.platforms = set(platforms)
+        self.build_kwargs = build_kwargs
+        self.osbs_client_config = osbs_client_config
+
+        self.worker_builds = []
+
+    def get_excluded_platforms(self):
+        df_dir = self.workflow.source.get_dockerfile_path()[1]
+        exclude_platforms = set()
+        exclude_path = os.path.join(df_dir, self.EXCLUDE_FILENAME)
+        if os.path.exists(exclude_path):
+            with open(exclude_path) as f:
+                for platform in f:
+                    platform = platform.strip()
+                    if not platform:
+                        continue
+                    exclude_platforms.add(platform)
+
+        return exclude_platforms
+
+    def get_platforms(self):
+        return self.platforms - self.get_excluded_platforms()
+
+    def get_current_builds(self, osbs):
+        field_selector = ','.join(['status!={status}'.format(status=status.capitalize())
+                                   for status in BUILD_FINISHED_STATES])
+        return len(osbs.list_builds(field_selector=field_selector))
+
+    def get_cluster_info(self, cluster, platform):
+        conf_file = os.path.join(self.osbs_client_config, 'osbs.conf')
+        osbs = OSBS(Configuration(conf_file=conf_file, conf_section=cluster.name),
+                    Configuration(conf_file=conf_file, conf_section=cluster.name))
+        current_builds = self.get_current_builds(osbs)
+        load = current_builds / cluster.max_concurrent_builds
+        return ClusterInfo(cluster, platform, osbs, load)
+
+    def choose_cluster(self, platform):
+        config = get_config(self.workflow)
+        clusters = [self.get_cluster_info(cluster, platform) for cluster in
+                    config.get_enabled_clusters_for_platform(platform)]
+
+        if not clusters:
+            raise RuntimeError('No clusters found for platform {}!'
+                               .format(platform))
+
+        selected = min(clusters, key=lambda c: c.load)
+        self.log.info('platform %s will use cluster %s',
+                      platform, selected.cluster.name)
+        return selected
+
+    def get_release(self):
+        labels = df_parser(self.workflow.builder.df_path, workflow=self.workflow).labels
```

Comment:
Yes! In this case, we actually care about the value of the release label, including env substitution from parent's env, as used by Fedora. Closed the other PR, as we need to pull the base image even for orchestrator build. Thank you for catching this.